app/controllers/questions_controller.rb

```diff
@@ -19,9 +23,111 @@ def show
       gon.solution = solution
     end
 
+    if !(current_user.show_solutions)
+      if @room.answered(@question.id,@room.id)
+        @number_of_question = @room.question_count - @room.question_count_not_visited(@room.id) + 1
+      else
+        @number_of_question = @room.question_count - @room.question_count_not_visited(@room.id)
+      end
+    else
+      @number_of_question = @room.learning_objects.where("learning_object_id < ?",@question.id).count + 1
+    end
+
+    @feedbacks = @question.feedbacks.includes(:user)
     @feedbacks = @question.feedbacks.visible.includes(:user)
   end
 
+  def eval_question
+    weight_solved = 5
+    weight_failed = 2
+    weight_dont_now = 1
+    @setup = Setup.take
+    @week = @setup.weeks.find_by_number(params[:week_number])
+
+    room = Room.find(params[:room_number])
+
+    lo_class = Object.const_get params[:type]
+    lo = lo_class.find(params[:id])
+
+    @results = UserToLoRelation.get_results_room(current_user.id,@week.id,params[:room_number], room.number_of_try)
+    result = @results.find {|r| r["result_id"] == lo.id.to_s}
```

Comment:
Odporucam dat ako parameter metody `get_results_room` aj ID learning_objectu a odfiltrovat tak tento learning_object uz pri komunikacii s databazou (a nerobit to cez metodu `find` dodatocne).